public/app/features/alerting/unified/components/receivers/form/GenerateAlertDataModal.tsx

```diff
@@ -0,0 +1,167 @@
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Stack } from '@grafana/experimental';
+import { Button, Card, Checkbox, Modal, useStyles2 } from '@grafana/ui';
+import { TestTemplateAlert } from 'app/plugins/datasource/alertmanager/types';
+
+import AnnotationsField from '../../rule-editor/AnnotationsField';
+import LabelsField from '../../rule-editor/LabelsField';
+
+interface Props {
+  isOpen: boolean;
+  onDismiss: () => void;
+  onAccept: (alerts: TestTemplateAlert[]) => void;
+}
+
+type AnnoField = {
+  key: string;
+  value: string;
+};
+
+interface FormFields {
+  annotations: AnnoField[];
+  labels: AnnoField[];
+  firing: boolean;
+}
+
+const defaultValues: FormFields = {
+  annotations: [{ key: '', value: '' }],
+  labels: [{ key: '', value: '' }],
+  firing: true,
+};
+
+export const GenerateAlertDataModal = ({ isOpen, onDismiss, onAccept }: Props) => {
+  const styles = useStyles2(getStyles);
+
+  const [alerts, setAlerts] = useState<TestTemplateAlert[]>([]);
+
+  const formMethods = useForm<FormFields>({ defaultValues, mode: 'onBlur' });
+  const annotations = formMethods.watch('annotations');
+  const labels = formMethods.watch('labels');
+  const firing = formMethods.watch('firing');
+
+  const onAdd = () => {
+    const alert: TestTemplateAlert = {
+      annotations: annotations
+        .filter(({ key, value }) => !!key && !!value)
+        .reduce((acc, { key, value }) => {
+          return { ...acc, [key]: value };
+        }, {}),
+      labels: labels
+        .filter(({ key, value }) => !!key && !!value)
+        .reduce((acc, { key, value }) => {
+          return { ...acc, [key]: value };
+        }, {}),
+      startsAt: '2023-04-01T00:00:00Z',
+      endsAt: firing ? addDays(new Date(), 1).toISOString() : addDays(new Date(), -1).toISOString(),
+    };
+    setAlerts((alerts) => [...alerts, alert]);
+    formMethods.reset();
+  };
+
+  const onSubmit = () => {
+    onAccept(alerts);
+    setAlerts([]);
+    formMethods.reset();
+  };
+
+  const labelsOrAnnotationsAdded = () => {
+    const someLabels = labels.some((lb) => lb.key !== '' && lb.value !== '');
+    const someAnnotations = annotations.some((ann) => ann.key !== '' && ann.value !== '');
+    return someLabels || someAnnotations;
+  };
+
+  return (
+    <Modal onDismiss={onDismiss} isOpen={isOpen} title={'Add alert data'}>
+      <FormProvider {...formMethods}>
+        <form
+          onSubmit={(e) => {
+            e.preventDefault();
+            e.stopPropagation();
+            formMethods.reset();
+          }}
+        >
+          <>
+            <Card>
+              <Stack direction="column" gap={1}>
+                <div className={styles.section}>
+                  <AnnotationsField />
+                </div>
+                <div className={styles.section}>
+                  <LabelsField />
+                </div>
+                <div className={styles.flexWrapper}>
+                  <Checkbox
```

Comment:
Maybe change this to a `<RadioButtonGroup>` to choose between `resoloved` and `firing` alert?

https://developers.grafana.com/ui/latest/index.html?path=/story/forms-radiobuttongroup--radio-buttons